src/plugins/data/server/search/strategies/esql_async_search/esql_async_search_strategy.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IScopedClusterClient, Logger } from '@kbn/core/server';
+import { catchError, tap } from 'rxjs/operators';
+import { getKbnServerError } from '@kbn/kibana-utils-plugin/server';
+import { SqlQueryRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { SqlGetAsyncResponse } from '@elastic/elasticsearch/lib/api/types';
+import { getKbnSearchError } from '../../report_search_error';
+import type { ISearchStrategy, SearchStrategyDependencies } from '../../types';
+import type { IAsyncSearchOptions } from '../../../../common';
+import { IKibanaSearchRequest, IKibanaSearchResponse, pollSearch } from '../../../../common';
+import { getDefaultAsyncGetParams, getDefaultAsyncSubmitParams } from './request_utils';
+import { toAsyncKibanaSearchResponse } from './response_utils';
+import { SearchConfigSchema } from '../../../../config';
+
+export const esqlAsyncSearchStrategyProvider = (
+  searchConfig: SearchConfigSchema,
+  logger: Logger,
+  useInternalUser: boolean = false
+): ISearchStrategy<
+  IKibanaSearchRequest<SqlQueryRequest['body']>,
+  IKibanaSearchResponse<SqlGetAsyncResponse>
+> => {
+  function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
+    const client = useInternalUser ? esClient.asInternalUser : esClient.asCurrentUser;
```

Comment:
`useInternalUser` shouldn't be necessary, it was only necessary for the default search strategy. (It can be removed from the non-async ESQL strategy as well, but that can be done separately if you'd like)

Reply:
i'll leave it for now